Drawing functions

lawruble13 opened the pull request:
Add ellipse, arc, and triangle functions.

lawruble13 committed:
Added arc and ellipse drawing functions

lawruble13 committed:
Added test for ellipse and arc

lawruble13 committed:
Add triangle functions

lawruble13 committed:
Merged drawing_functions with upstream changes

lawruble13 committed:
Pre-rebase to dev commit, working on triangle & tidying code

lawruble13 committed:
Added triangle, shape helpers, etc

lawruble13 committed:
Merge to 0.0.12; add triangle, arc, and ellipse; version bump

lawruble13 committed:
Added triangle test

AlphaRLee commented on spark/core.py:
```diff
@@ -310,32 +314,118 @@ def clear_rect(self, *args):
-        self.canvas.stroke_arc(*arc_args)
-        
+        self.check_coords("stroke_circle", *args, width_only = True)
+        self.fill_ellipse(*args, args[2])
```
Is this supposed to be `stroke_ellipse`?
```suggestion
        self.stroke_ellipse(*args, args[2])
```

lawruble13 replied:
Yep, good catch! A failure on my part to not check the individual circle functions thoroughly.

AlphaRLee commented on spark/core.py:
```diff
@@ -310,32 +314,118 @@ def clear_rect(self, *args):
+        self.fill_ellipse(*args, args[2])
+
+    def ellipse(self, *args):
+        self.check_coords("ellipse", *args)
```
Any performance concerns over these redundant `check_coords` calls (between `circle` -> `ellipse` -> `fill_ellipse` -> `fill_arc`)?
We could optimize it by adding a hidden parameter/flag that disables these validation checks, but it may throw monkey wrenches in our logic.

lawruble13 replied:
As `check_coords` is only a check on the correct number and type, I don't think it'll have a super significant performance increase, but if it can be optimized simply that'd be great.

lawruble13 committed:
Fix stroke_circle

Yep, sure is! Forgot to test the circle functions after the last change to them, thanks for the catch

Co-authored-by: R Lee <alpha.r.lee.dev@gmail.com>

raduschirliu committed:
Merged latest dev
